core-api/src/test/java/com/optimizely/ab/OptimizelyTest.java

```diff
@@ -2291,22 +2166,33 @@ public void trackEventWithNoValidExperiments() throws Exception {
             eventType = validProjectConfig.getEventNameMapping().get("clicked_purchase");
         }
 
-        when(mockDecisionService.getVariation(any(Experiment.class), any(String.class), anyMapOf(String.class, String.class)))
-                .thenReturn(null);
+        // setup a mock event builder to return expected conversion params
+        EventFactory mockEventFactory = mock(EventFactory.class);
+
         Optimizely optimizely = Optimizely.builder(validDatafile, mockEventHandler)
-                .withDecisionService(mockDecisionService)
+                .withEventBuilder(mockEventFactory)
+                .withConfig(validProjectConfig)
                 .build();
 
         Map<String, String> attributes = new HashMap<String, String>();
         attributes.put("browser_type", "firefox");
 
-        logbackVerifier.expectMessage(Level.INFO,
-                "There are no valid experiments for event \"" + eventType.getKey() + "\" to track.");
-        logbackVerifier.expectMessage(Level.INFO, "Not tracking event \"" + eventType.getKey() +
-                "\" for user \"userId\".");
-        optimizely.track(eventType.getKey(), testUserId, attributes);
+        when(mockEventFactory.createConversionEvent(
+            eq(validProjectConfig),
+            eq(genericUserId),
+            eq(eventType.getId()),
+            eq(eventType.getKey()),
+            eq(attributes),
+            eq(Collections.<String, Object>emptyMap())))
+            .thenReturn(logEventToDispatch);
 
-        verify(mockEventHandler, never()).dispatchEvent(any(LogEvent.class));
+        logbackVerifier.expectMessage(Level.INFO, "Tracking event \"" + eventType.getKey() +
+            "\" for user \"" + genericUserId + "\".");
+        logbackVerifier.expectMessage(Level.DEBUG, "Dispatching conversion event to URL test_url with params " +
+            testParams + " and payload \"{}\"");
+
+        optimizely.track(eventType.getKey(), genericUserId, attributes);
+        verify(mockEventHandler).dispatchEvent(eq(logEventToDispatch));
```

Comment:
minor: since we are mocking the event factory (for better or worse), maybe consider [`sameInstance()`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/CoreMatchers.html#sameInstance(T)) instead of `eq()` for a little more stringency. also can consider using `verifyNoMoreInteractions(mockEventFactory)` if that's what we expect. if you decide to incorporate, could probably do in a helper method on this test class, as this isn't the only test it could be applied to.